core/App/ehsm_napi.cpp

```diff
@@ -1125,4 +1129,107 @@ char* NAPI_ExportDataKey(const char* cmk_base64,
     return retJsonObj.toChar();
 }
 
-}  // extern "C"
+/*
+@return
+[string] json string
+    {
+        code: int,
+        message: string,
+        result: {
+            challenge_base64 : string,
+            ga_base64 : string
+        }
+    }
+*/
+char *NAPI_RA_HANDSHAKE_MSG0(const char *p_msg0)
+{
+    print_d("***NAPI_RA_HANDSHAKE_MSG0 start.");
+
+    print_d("msg0: \n %s", p_msg0);
+    std::string challenge_base64 = "challenge_base64";
+    std::string ga_base64 = "ga_base64";
```

Comment:
not implemented yet?

Reply:
this patch will not include the RA protocol.

Reply:
yes